src/proxy.rs

```diff
@@ -407,6 +410,37 @@ pub async fn relay(
     }
 }
 
+// pick_upstream_service selects an upstream service for inbound metrics.
+// There may be many services for a single workload. We find the the first one with an applicable port
+// as a best guess.
+pub fn pick_upstream_service(
+    conn: &Connection,
+    upstream_service: Vec<Service>,
+    dest: &Workload,
+) -> Option<ServiceDescription> {
+    let dport = conn.dst.port();
+    error!("upstream service... {upstream_service:?}");
+    let netaddr = network_addr(&dest.network, conn.dst.ip());
+    let euid = endpoint_uid(&dest.uid, Some(&netaddr));
+    upstream_service
+        .iter()
+        .find(|s| {
+            for (sport, tport) in s.ports.iter() {
+                if tport == &dport {
+                    // TargetPort directly matches
+                    return true;
+                }
+                if s.endpoints.get(&euid).and_then(|e| e.port.get(sport)) == Some(&dport) {
```

Comment:
can you clarify when endpoint port is different from service port

Reply:
yes added a comment. tl;dr Named port in k8s